README.md

<img src="icons/365764486_123125147525866_6875513157502364595_n.png" dir="rtl" align="center">

<p dir="rtl" align="right">السلام عليكم ورحمة الله وبركاتة </p>

<p dir="rtl" align="right">
اضافة للمتصفح للمساعدة في ايجاد او الوصول الي بعض المسارات المهمة او المعلومات المهمة من خلال الصفحة وملفات الجافا سكربت والاداة تعمل بشكل سلبي بمعني انها لا ترسل اي طلبات للموقع جديدة.
</p>

## <h2 dir="rtl" align="right"> هدفنا</h2>
<p dir="rtl" align="right">هدفنا هو محاولة تعريب المحتوي الاجنبي </p>

<p dir="rtl" align="right">
حسابي علي موقع اكس او تويتر:
https://twitter.com/0xElkomy
</p>

## <h2 dir="rtl" align="right"> التثبيت google chrome</h2>
<p dir="rtl" align="right">
هتنزل الفولد وتنخسه تحطه في 
Extenions Page
عندك زي كدا
</p>

![Chrome](image.png)

<p dir="rtl" align="right">
تنسخه وتحطه في البيدج دي 
</p>

- brave://extensions/
- chrome://extensions/

<p dir="rtl" align="right">
او تعمل 
load unpacked زي الصورة الي فوق
</p>


## <h2 dir="rtl" align="right"> نسخة فايرفوكس</h2>

https://addons.mozilla.org/en-US/firefox/addon/findsomething-%D8%A7%D9%8A%D8%AC%D8%A7%D8%AF-%D8%B4%D8%A6-%D9%85%D8%A7/


## <h2 dir="rtl" align="right"> شكرا</h2>
@momosecurity


CHANGELOG.md
<p dir="rtl" align="right">

# سجل التغييرات
سيتم تسجيل جميع التغييرات البارزة لهذا المشروع في هذا الملف.

## [2.0.16]-2023-07-09
شكراً لـ "الرياح الطويلة" على التغذية الراجعة حول مشكلة عدم توفر معلومات رد webhook بالكامل.

### التحسينات
- تم تحسين هيكل التحديث في الخلفية.

### الإصلاحات
- تم إصلاح مشكلة عدم توفر معلومات رد webhook بالكامل.

## [2.0.15]-2023-06-26
شكراً لـ "الانتظار الفردي" و"مطور Wild Code" و"MiaoTony" و"KexinCC" على التغذية الراجعة حول مشكلة عدم القدرة على تسجيل الدخول.
شكراً لـ "Xc1Ym" على التغذية الراجعة حول مشكلة القائمة البيضاء.

### الإصلاحات
- تم إصلاح مشكلة عدم القدرة على تسجيل الدخول إلى بعض المواقع. سبب هذه المشكلة هو أن الإضافة كانت تقوم بطلب الصفحة مرة أخرى بعد فتحها، مما يؤدي إلى انتهاء صلاحية رمز token لمرة واحدة. الآن تم استبدال ذلك بإستخدام استخراج المعلومات مباشرة من مصدر الصفحة بدلاً من إعادة الطلب.
- تم إصلاح خطأ في القائمة البيضاء، الآن يمكن استخدام القائمة البيضاء لتطبيق القيود على النطاق (IP) أو النطاق (IP) + المنفذ، مثل a.com سيعمل على sub.a.com و a.com:8001.

## [2.0.14]-2023-03-15
شكراً لـ "hyasin" على التغذية الراجعة حول تجربة نسخ النص.

### التحسينات
- تم نقل تلميحات المصدر للمعلومات إلى أعلى الأيقونة <، لتحسين تجربة نسخ المعلومات.
- تم تحسين منطق استخراج معلومات IP واسم النطاق، لاستخراج المزيد من المعلومات ذات الصلة من الروابط التي تم استخراجها مسبقًا.

### الإصلاحات
- تم إصلاح خطأ في عرض القائمة البيضاء بعد تحديد إعدادات القائمة البيضاء.

## [2.0.13]-2023-03-07
شكراً لـ "osxtest" على المساهمة في هذا المشروع.
شكراً لـ "بطولة" و"مطور Wild Code" على التغذية الراجعة حول مشكلة تسجيل الدخول.
شكراً لـ "مطور وحيد" على التغذية الراجعة حول مشكلة القائمة البيضاء.

### الإضافات
- تم إضافة نصائح تلميح المصدر والروابط، حيث يمكنك الآن فتح رابط المصدر للمعلومات عن طريق "ctrl + النقر بزر الفأرة الأيسر" أو "النقر بزر الفأرة الأيمن وفتح الرابط في علامة تبويب جديدة".

### الإصلاحات
- تم إصلاح بعض الأخطاء في التجربة.

## [2.0.12]-2023-02-14
شكراً لـ "صندوق صغير" على التنبيه عن الأخطاء.

### الإصلاحات
- تم إصلاح مشكلة عدم وجود علامة النص عندما لا يكون هناك علامات script في الصفحة.
- تم إصلاح بعض الأخطاء الناتجة عن التحقق من القيم الفارغة.

## [2.0.11]-2022-11-09
شكراً لـ "xiaoliangli1128" على التنبيه عن الأخطاء وتقديم بعض القواعد.

### الإضافات
- تم إضافة بعض قواعد استخراج المعلومات الحساسة.

### الإصلاحات
- تم إصلاح مشكلة استخراج غير كامل للمعلومات الحساسة، بسبب استخدام متغيرات مكررة.

## [2.0.10]-2022-08-27

### الإصلاحات
- تم إصلاح مشكلة عدم القدرة على نسخ البيانات من النافذة العائمة العالمية في الإصدار 2.0.9.

## [2.0.9]-2022-08-26

### الإضافات
- تم إضافة تكوين الانتهاء التلقائي، بعد 2 ثانية من عدم الاستجابة سيتم إلغاء الطلب، لتجنب الانتظار الطويل.
- تم إضافة وظيفة الاكمال التلقائي للمسار عند النسخ.

### التغييرات
- تم تعديل ارتفاع النافذة العائمة العالمية، حيث كانت طويلة جداً.

### الإصلاحات
- تم إصلاح مشكلة فقدان البيانات بعد توقف النص الخلفي للتشغيل، من خلال حفظ البيانات المستخرجة في localstorage.
- تم إصلاح بعض الأخطاء المتعلقة بتجربة المستخدم.

## [2.0.8]-2022-06-18
شكراً لـ "mdkk" على الاقتراحات في هذا الإصدار.
شكراً لـ "NeeSung" على الاقتراحات بشأن واجهة المستخدم.

### الإضافات
- تم إضافة صفحة تكوين webhook، عند اكتمال معالجة الخدمة الخلفية، ستتم إرسال البيانات إلى API محدد.
- تم إضافة نافذة عائمة عالمية يمكن سحبها، يمكن التحكم فيها من صفحة التكوين.
- تم إضافة عرض لتقدم المعالجة.

### التغييرات
- تم تحديث رقم الإصدار في ملف التكوين إلى 3.
- تم تحسين واجهة المستخدم.

### الإصلاحات
- تم محاولة إصلاح بعض مشاكل الاستخراج غير المكتمل للبيانات.

## [2.0.7]-2022-05-26
شكراً لـ "0cat" على اقتراحات هذا الإصدار.

### الإضافات
- تم إضافة استخراج بعض البيانات الحساسة باستخدام الكلمات الرئيسية، واستخدمت قواعد nuclei، شكراً للمصدر المفتوح. https://github.com/projectdiscovery/nuclei-templates/blob/master/exposures/tokens/generic/credentials-disclosure.yaml
- تم إضافة وظيفة النسخ السريع، حيث يمكنك النقر فوق الزر لنسخ نتائج المنطقة بشكل فوري.
- تم إضافة استخراج مسار بدء غير قياسي، وقد يؤدي ذلك إلى بعض التنبيهات الزائفة. على سبيل المثال، api/path.

### التغييرات
- تم نقل استخراج البيانات إلى الخلفية لتجنب تعطيل تحميل الصفحة الأمامية.

### الإصلاحات
- تم محاولة إصلاح بعض مشاكل استخراج البيانات.

## [2.0.6]-2022-05-03

### التغييرات
- تم تحسين الكود العادي، واضافة ./ و ../ للمسارات وأيضاً الاطلاع على المنافذ في العناوين الرابطة.

## [2.0.5]-2022-01-07

### التغييرات
- تم إضافة علامات اقتباس فردية لاستخراج الروابط.
- تم الاحتفاظ بالروابط النصية داخل علامات script بدون انتهاء بـ .js.

## [2.0.1]-2021-06-26

### الإضافات
- تم إضافة مطابقة الوظائف الخاصة بالتشفير/التوقيع، حيث يكون الخوارزميات المطابقة هي base64(base64، btoa، atob)، aes(CryptoJS.AES)، des(CryptoJS.DES)، rsa(JSEncrypt، rsa، KJUR)، md5(md5)، sha1(sha1)، sha256(sha256)، sha512(sha512).
### التغييرات
- تم تحديث بعض قواعد التعبير العادية للبطاقة الشخصية ورقم الهاتف والبريد الإلكتروني ورمز JWT، شكراً جزيلاً للقواعد المفتوحة المصدر. https://gh0st.cn/HaE/

## [2.0.0]-2020-09-12

### الإضافات
- تم إضافة مطابقة للعنوان الرقمي (IP)، العنوان الرقمي (IP) + المنفذ، النطاق، المسار، الرابط، المسار الثابت، بطاقة الهوية، رقم الهاتف، البريد الإلكتروني، وغيرها من الحقول وعرضها.

</p>


popup.js
// @Date    : 2020-09-12 16:26:48
// @Author  : residuallaugh
var key = ["ip","ip_port","domain","path","incomplete_path","url","static","sfz","mobile","mail","jwt","algorithm","secret"]

function init_copy() {
    var elements = document.getElementsByClassName("copy");
    if(elements){
        for (var i=0, len=elements.length|0; i<len; i=i+1|0) {
            let ele_name = elements[i].name;
            let ele_id = elements[i].id;
            elements[i].onclick=function () {
                var inp =document.createElement('textarea');
                document.body.appendChild(inp)
                var copytext = document.getElementById(ele_name).textContent;
                if (ele_id == "path_url_button"){
                    Promise.all([getCurrentTab().then(function x(tab){
                        // console.log(tab);
                        if(tab == undefined){
                            alert("الرجاء النقر فوق الصفحة الأصلية ثم نسخ: ")
                            return;
                        }
                        var url = new URL(tab.url)
                        var path_list = copytext.split('\n')
                        copytext = ""
                        for (var i = path_list.length - 1; i >= 0; i--) {
                            if(path_list[i][0] == '.'){
                                copytext += url.origin+'/'+path_list[i]+'\n';
                            }else{
                                copytext += url.origin+path_list[i]+'\n';
                            }
                        }
                        inp.value = copytext.slice(0, -1);
                        inp.select();
                        // console.log(copytext)
                        document.execCommand('copy',false);
                        // inp.remove();
                    })]).then(res=> inp.remove())
                    // alert('نسخ بنجاح');
                    return ;
                }
                inp.value = copytext;
                inp.select();
                document.execCommand('copy',false);
                inp.remove();
                // alert('نسخ بنجاح'');
            }
        }
    }
};


async function getCurrentTab() {
  let queryOptions = { active: true, lastFocusedWindow: true };
  let [tab] = await chrome.tabs.query(queryOptions);
  return tab;
}
function sleep (time) {
  return new Promise((resolve) => setTimeout(resolve, time));
}

function show_info(result_data) {
    for (var k in key){
        if (result_data[key[k]]){
            // console.log(result_data[key[k]])
            let container = document.getElementById(key[k]);
            while((ele = container.firstChild)){
                ele.remove();
            }
            container.style.whiteSpace = "pre";
            for (var i in result_data[key[k]]){
                let tips = document.createElement("div");
                tips.setAttribute("class", "tips")
                let link = document.createElement("a");
                let source = result_data['source'][result_data[key[k]][i]];
                if (source) {
                    //على الرغم من أنه من المستحيل تجنب كونه xss ، فإن المكون الإضافي يوفر CSP الصحيح بشكل افتراضي ، مما يعني أن جافا سكريبت لن يتم تنفيذها حتى لو لم نتعامل معها بشكل خاص.
                    // source = 'javascript:console.log`1`'
                    link.setAttribute("href", source);
                    link.setAttribute("title", source);
                }
                link.appendChild(tips);
                let span = document.createElement("span");
                span.textContent = result_data[key[k]][i]+'\n';
                container.appendChild(link);
                container.appendChild(span);
            }
        }
    }
}

getCurrentTab().then(function get_info(tab) {
    // console.log("findsomething_result_"+tab.url)
    chrome.storage.local.get(["findsomething_result_"+tab.url], function(result_data) {
        if (!result_data){
            sleep(100);
            get_info(tab);
            return;
        }
        result_data = result_data["findsomething_result_"+tab.url]
        // console.log(result_data)
        if(!result_data || result_data['done']!='done'){
            // console.log('لم يتم استخراجه بعد');
            if(result_data && result_data.donetasklist){
                // console.log("findsomething_result_"+tab.url)
                chrome.storage.local.get(["findsomething_result_"+tab.url], function(result){show_info(result["findsomething_result_"+tab.url]);});
                // show_info(result_data);
                document.getElementById('taskstatus').textContent = "تتم المعالجة الان"+result_data['donetasklist'].length+"/"+result_data['tasklist'].length;
            }else{
                document.getElementById('taskstatus').textContent = "تتم المعالجة الان";
            }
            sleep(100);
            get_info(tab);
            return;
        }
        // console.log(result_data)
        document.getElementById('taskstatus').textContent = "اكتملت المعالجة: "+result_data['donetasklist'].length+"/"+result_data['tasklist'].length;
        chrome.storage.local.get(["findsomething_result_"+tab.url], function(result){show_info(result["findsomething_result_"+tab.url]);});
        // show_info(result_data);
        // النتائج غير متسقة وتستمر في التحديث
        // if(result_data['donetasklist'].length!=result_data['tasklist'].length){
        //     get_info(tab);
        // }
        return;
    });
});


init_copy();

settings.js

document.getElementById("save").onclick=function () {
	let webhook_setting = {};
	webhook_setting['url'] = document.getElementById('url').value;
	webhook_setting['method'] = document.getElementById('method').value;
	webhook_setting['arg'] = document.getElementById('arg').value;
	webhook_setting['headers'] = JSON.parse(document.getElementById('headers').value);
	// console.log(webhook_setting);
	chrome.storage.local.set({"webhook_setting": webhook_setting});
}
document.getElementById("reset").onclick=function () {
	let webhook_setting = {"url":"","arg":"","headers":{}};
	document.getElementById('url').value = "";
	document.getElementById('arg').value = "";
	document.getElementById('headers').value = "{}";
	// console.log(webhook_setting);
	chrome.storage.local.set({"webhook_setting": webhook_setting});
}

document.getElementById("save_allowlist").onclick=function () {
	snsArr = document.getElementById('allowlist').value.split(/[(\r\n)\r\n]+/);
	snsArr.forEach((item, index)=>{
		if(!item){
			snsArr.splice(index,1);
		}
	})
	// console.log(snsArr)
	chrome.storage.local.set({"allowlist": snsArr});
}
document.getElementById("reset_allowlist").onclick=function () {
	document.getElementById('allowlist').value = "";
	chrome.storage.local.set({"allowlist": []});
}

document.getElementById("global_float").onclick=function () {
	// var webhook_setting = {};
	chrome.storage.local.get(["global_float"], function(settings){
		// console.log(settings);
		chrome.storage.local.set({"global_float": settings["global_float"]==true ? false : true});
		document.getElementById('global_float').textContent = settings["global_float"]==true ? "مغلق" : "مفتوح";
	});
	// console.log(webhook_setting);
}

document.getElementById("fetch_timeout").onclick=function () {
	// var webhook_setting = {};
	chrome.storage.local.get(["fetch_timeout"], function(settings){
		// console.log(settings);
		chrome.storage.local.set({"fetch_timeout": settings["fetch_timeout"]==true ? false : true});
		document.getElementById('fetch_timeout').textContent = settings["fetch_timeout"]==true ? "مغلق" : "مفتوح";
	});
	// console.log(webhook_setting);
}

chrome.storage.local.get(["webhook_setting"], function(settings){
	console.log(settings);
	if(!settings || settings == {} || !settings["webhook_setting"] ){
        console.log('webhook_اعدادت');
        return;
    }
	document.getElementById('url').value = settings["webhook_setting"]['url'];
	document.getElementById('method').value = settings["webhook_setting"]['method'];
	document.getElementById('arg').value = settings["webhook_setting"]['arg'];
	document.getElementById('headers').value = JSON.stringify(settings["webhook_setting"]['headers']);
});
chrome.storage.local.get(["global_float"], function(settings){
	document.getElementById('global_float').textContent = settings["global_float"]==true ? "مفتوح" : "مغلق";
});
chrome.storage.local.get(["fetch_timeout"], function(settings){
	document.getElementById('fetch_timeout').textContent = settings["fetch_timeout"]==true ? "مفتوح" : "مغلق";
});
chrome.storage.local.get(["allowlist"], function(allowlist){
	if(allowlist && allowlist["allowlist"]){
		document.getElementById('allowlist').textContent = allowlist["allowlist"].join('\n');
	}
});


content.js
// @Date    : 2020-09-12 16:26:48
// @Author  : residuallaugh
(function(){
    var protocol = window.location.protocol;
    var host = window.location.host;
    var domain_host = host.split(':')[0];
    var href = window.location.href;
    // var source = document.getElementsByTagName('html')[0].innerHTML;
    var source = document.documentElement.outerHTML;
    var hostPath;
    var urlPath;
    var urlWhiteList = ['.google.com','.amazon.com','portswigger.net','bing.com','hackerone.com','facebook.com'];
    var target_list = [];
    var source_href = source.match(/href=['"].*?['"]/g);
    var source_src = source.match(/src=['"].*?['"]/g);
    var script_src = source.match(/<script [^><]*?src=['"].*?['"]/g);
    chrome.storage.local.get(["allowlist"], function(settings){
        // console.log(settings , settings['allowlist'])
        if(settings && settings['allowlist']){
            urlWhiteList = settings['allowlist'];
        }
        for(var i = 0;i < urlWhiteList.length;i++){
            if(host.endsWith(urlWhiteList[i]) || domain_host.endsWith(urlWhiteList[i])){
                console.log('اسم المجال في القائمة البيضاء ، تخطي الصفحة الحالية')
                return ;
            }
        }
        // target_list.push(window.location.href);
        
        // console.log(source_href,source_src,script_src)
        if(source_href){
            for(var i=0;i<source_href.length;i++){
                var u = deal_url(source_href[i].substring(6,source_href[i].length-1));
                if(u){
                    target_list.push(u);
                }
            }
        }
        if(source_src){
            for(var i=0;i<source_src.length;i++){
                var u = deal_url(source_src[i].substring(5,source_src[i].length-1));
                if(u){
                    target_list.push(u);
                }
            }
        }
        
        const tmp_target_list=[];
        for (var i = 0;i<target_list.length;i++){
            if (tmp_target_list.indexOf(target_list[i])===-1){
              tmp_target_list.push(target_list[i])
            }
        }
        tmp_target_list.pop(href)
        chrome.runtime.sendMessage({greeting: "find",data: target_list, current: href, source: source});
    });
    function is_script(u){
        if(script_src){
            for(var i=0;i<script_src.length;i++){
                if (script_src[i].indexOf(u)>0){
                    return true
                }
            }
        }
        return false
    }

    function deal_url(u){
        if(u.indexOf(".js")=='-1' && !is_script(u)){
            return ;
        }else if(u.substring(0,4)=="http"){
            return u;
        }
        else if(u.substring(0,2)=="//"){
            return protocol+u;
        }
        else if(u.substring(0,1)=='/'){
            return protocol+'//'+host+u;
        }
        else if(u.substring(0,2)=='./'){
            if (href.indexOf('#')>0) {
                tmp_href = href.substring(0,href.indexOf('#'))
            }else{
                tmp_href = href;
            }
            return tmp_href.substring(0,tmp_href.lastIndexOf('/')+1)+u;
        }else{
            // console.log("not match prefix:"+u+",like http // / ./")
            if (href.indexOf('#')>0) {
                tmp_href = href.substring(0,href.indexOf('#'))
            }else{
                tmp_href = href;
            }
            return tmp_href.substring(0,tmp_href.lastIndexOf('/')+1)+u;
        }
    }
    

})()


chrome.storage.local.get(["global_float"], function(settings){
    // console.log(settings);
    if (settings["global_float"]!=true){
        return
    }
    // console.log(settings["global_float"]);
    // console.log("findsomething-divglobal_float");
    // استخدم تسميات مخصصة
    const body = document.getElementsByTagName('html')[0];
    const div = document.createElement('div');
    div.setAttribute("id","findsomething-float-div");
    div.innerHTML = `
    <findsomething-div dir="rtl" align="right" id="findsomething_neko" style="width:410px;max-height:500px;font-size:14px;color:#000000;box-shadow: 0 2px 12px 0 rgba(0,0,0,0.1) ;background-color: #fff;border-radius: 5px;border: 1px solid #ebebeb;right:20px;top:20px;position: fixed;z-index: 1000000;overflow:scroll;">
          <findsomething-div id="findsomething_neko-title" style="display: flex;justify-content: space-between;">
            <findsomething-div id="findsomething_taskstatus" style="height: 34px; line-height: 34px; margin-right: 10px;"></findsomething-div>
            <findsomething-div style="cursor: pointer;margin-top: 2px;margin-right: 10px;" onclick='(function(){document.getElementById("findsomething-float-div").removeChild(document.getElementById("neko"));})()'>隐藏</findsomething-div>
          </findsomething-div>
            <findsomething-div style="width: 300px; margin-top: 10px;">
                <findsomething-div class="findsomething-title">IP<button type="button" class="finsomething_copy" name="ip"><p dir="rtl" align="right">نسخ</p></button></findsomething-div>
                <findsomething-p id="findsomething_ip" style="word-break:break-word;margin-right:10px;">🚫</findsomething-p>
                <findsomething-div class="findsomething-title"><p dir="rtl" align="right">بورتات الايبي</p><button class="findsomething_copy" name="ip_port"><p dir="rtl" align="right">نسخ</p></button></findsomething-div>
                <findsomething-p id="findsomething_ip_port" style="word-break:break-word;margin-right:10px;">🚫</findsomething-p>
                <findsomething-div class="findsomething-title"><p dir="rtl" align="right">اسم الدومين </p><button class="findsomething_copy" name="domain"><p dir="rtl" align="right">نسخ</p></button></findsomething-div>
                <findsomething-p id="findsomething_domain" style="word-break:break-word;margin-right:10px;">🚫</findsomething-p>
                <findsomething-div class="findsomething-title"><p dir="rtl" align="right">بطاقة التعريف</p><button class="findsomething_copy" name="sfz"><p dir="rtl" align="right">نسخ</p></button></findsomething-div>
                <findsomething-p id="findsomething_sfz" style="word-break:break-word;margin-right:10px;">🚫</findsomething-p>
                <findsomething-div class="findsomething-title"><p dir="rtl" align="right">رقم التليفون</p><button class="findsomething_copy" name="mobile"><p dir="rtl" align="right">نسخ</p></button></findsomething-div>
                <findsomething-p id="findsomething_mobile" style="word-break:break-word;margin-right:10px;">🚫</findsomething-p>
                <findsomething-div class="findsomething-title"><p dir="rtl" align="right">بريد الالكتروني </p><button class="findsomething_copy" name="mail"><p dir="rtl" align="right">نسخ</p></button></findsomething-div>
                <findsomething-p id="findsomething_mail" style="word-break:break-word;margin-right:10px;">🚫</findsomething-p>
                <findsomething-div class="findsomething-title">JWT<button class="findsomething_copy" name="jwt"><p dir="rtl" align="right">نسخ</p></button></findsomething-div>
                <findsomething-p id="findsomething_jwt" style="word-break:break-word;margin-right:10px;">🚫</findsomething-p>
                <findsomething-div class="findsomething-title"><p dir="rtl" align="right">الخوارزميات </p><button class="findsomething_copy" name="algorithm"><p dir="rtl" align="right">نسخ</p></button></findsomething-div>
                <findsomething-p id="findsomething_algorithm" style="word-break:break-word;margin-right:10px;">🚫</findsomething-p>
                <findsomething-div class="findsomething-title"><p dir="rtl" align="right">اشياء سرية تقريبا</p><button class="findsomething_copy" name="secret"><p dir="rtl" align="right">نسخ</p></button></findsomething-div>
                <findsomething-p id="findsomething_secret" style="word-break:break-word;margin-right:10px;">🚫</findsomething-p>
                <findsomething-div class="findsomething-title"><p dir="rtl" align="right">المسار</p><button class="findsomething_copy" name="path"><p dir="rtl" align="right">نسخ</p></button></findsomething-div>
                <findsomething-p id="findsomething_path" style="word-break:break-word;margin-right:10px;">🚫</findsomething-p>
                <findsomething-div class="findsomething-title"><p dir="rtl" align="right">المسار غير مكتمل </p><button class="findsomething_copy" name="incomplete_path"><p dir="rtl" align="right">نسخ</p></button></findsomething-div>
                <findsomething-p id="findsomething_incomplete_path" style="word-break:break-word;margin-right:10px;">🚫</findsomething-p>
                <findsomething-div class="findsomething-title">Url<button class="findsomething_copy" name="url"><p dir="rtl" align="right">نسخ</p></button></findsomething-div>
                <findsomething-p id="findsomething_url" style="word-break:break-word;margin-right:10px;">🚫</findsomething-p>
                <findsomething-div class="findsomething-title">StaticUrl<button class="findsomething_copy" name="static"><p dir="rtl" align="right">نسخ</p></button></findsomething-div>
                <findsomething-p id="findsomething_static" style="word-break:break-word;margin-right:10px;">🚫</findsomething-p>
            </findsomething-div>
    </findsomething-div>
        <style type="text/css">
        .findsomething_copy {
            border-style: none;
            background-color: #ffffff;
            float: right;
            margin-right: 0px;
            font-size: 14px;
        }
        findsomething-div{
            display: block;
        }
        findsomething-p {
            display: block;
            margin-top: 14px;
            margin-bottom: 14px;
            line-height: 14px;
        }

        .findsomething-title {
            font-size: 16px;
            font-weight: bold;
            border-right: 4px solid black;
            text-indent: 4px;
            height: 16px;
            line-height: 16px;
            width: 100%;
            margin-right: 10px;
        }

        button{
            cursor: pointer
        }
        </style>
        `
    body.appendChild(div)
    var neko = document.querySelector('#findsomething_neko');
    var nekoW = neko.offsetWidth;
    var nekoH = neko.offsetHeight;
    var cuntW = 0;
    var cuntH = 0;
    neko.style.right = parseInt(document.body.offsetWidth - nekoW)+1 + 'px';
    neko.style.top = '50px';
    move(neko, 0, 0);
    function move(obj, w, h) {
        if (obj.direction === 'right') {
            obj.style.right = 0 - w + 'px';
        } else if (obj.direction === 'right') {

            obj.style.right = document.body.offsetWidth - nekoW + w + 'px';
        }
        if (obj.direction === 'top') {
            obj.style.top = 0 - h + 'px';
        } else if (obj.direction === 'bottom') {
            obj.style.top = document.body.offsetHeight - nekoH + h + 'px';
        }
    }

    function rate(obj, a) {
        //  console.log(a);
        obj.style.transform = ' rotate(' + a + ')'
    }

    var nekotitle = document.querySelector('#findsomething_neko-title');
    nekotitle.onmousedown = function (e) {
        var nekoL = e.clientX - neko.offsetright;
        var nekoT = e.clientY - neko.offsetTop;
        document.onmousemove = function (e) {
            cuntW = 0;
            cuntH = 0;
            neko.direction = '';
            neko.style.transition = '';
            neko.style.right = (e.clientX - nekoL) + 'px';
            neko.style.top = (e.clientY - nekoT) + 'px';
            if (e.clientX - nekoL < 5) {
                neko.direction = 'right';
            }
            if (e.clientY - nekoT < 5) {
                neko.direction = 'top';
            }
            if (e.clientX - nekoL > document.body.offsetWidth - nekoW - 5) {
                neko.direction = 'right';
            }
            if (e.clientY - nekoT > document.body.offsetHeight - nekoH - 5) {
                neko.direction = 'bottom';
            }

            move(neko, 0, 0);


        }
    }
    neko.onmouseover = function () {
        move(this, 0, 0);
        rate(this, 0)
    }

    neko.onmouseout = function () {
        // move(this, nekoW / 2, nekoH / 2);
        move(this, nekoW / 2, 0);
        // move(this, 0, 0);
    }

    neko.onmouseup = function () {
        document.onmousemove = null;
        this.style.transition = '.5s';
        // move(this, nekoW / 2, nekoH / 2);
        move(this, nekoW / 2, 0);
    }

    window.onresize = function () {
        var bodyH = document.body.offsetHeight;
        var nekoT = neko.offsetTop;
        var bodyW = document.body.offsetWidth;
        var nekoL = neko.offsetright;

        if (nekoT + nekoH > bodyH) {
            neko.style.top = bodyH - nekoH + 'px';
            cuntH++;
        }
        if (bodyH > nekoT && cuntH > 0) {
            neko.style.top = bodyH - nekoH + 'px';
        }
        if (nekoL + nekoW > bodyW) {
            neko.style.right = bodyW - nekoW + 'px';
            cuntW++;
        }
        if (bodyW > nekoL && cuntW > 0) {
            neko.style.right = bodyW - nekoW + 'px';
        }

        // move(neko, nekoW / 2, nekoH / 2);
        move(this, nekoW / 2, 0);
        // move(this, 0, 0);
    }
});

function init_copy() {
    var elements = document.getElementsByClassName("findsomething_copy");
    if (elements) {
        for (var i=0, len=elements.length|0; i<len; i=i+1|0) {
            let ele_name = elements[i].name;
            elements[i].onclick=function () {
                // console.log('copy begin');
                var inp =document.createElement('textarea');
                document.body.appendChild(inp)
                inp.value =document.getElementById(ele_name).textContent;
                inp.select();
                document.execCommand('copy',false);
                inp.remove();
                // console.log('copy end');
            }
        }
    }
};
setTimeout(()=>{
    init_copy();
}, 500);

function sleep (time) {
  return new Promise((resolve) => setTimeout(resolve, time));
}

var key = ["ip","ip_port","domain","path","incomplete_path","url","static","sfz","mobile","mail","jwt","algorithm","secret"]

function show_info(result_data) {
    if(result_data){
        for (var k in key){
            if (result_data[key[k]]){
                // console.log(result_data[key[k]])
                let p="";
                for(var i in result_data[key[k]]){
                    p = p + result_data[key[k]][i] +'\n'
                }
                document.getElementById("findsomething_"+key[k]).style.whiteSpace="pre";
                document.getElementById("findsomething_"+key[k]).textContent=p;
            }
        }
    }
}
function get_info() {
    chrome.runtime.sendMessage({greeting: "get", current: window.location.href}, function(result_data) {
        let taskstatus = document.getElementById('findsomething_taskstatus');
        if(!taskstatus){
            return;
        }
        if(!result_data|| result_data['done']!='done'){
            // console.log('还未提取完成');
            if(result_data){
                show_info(result_data);

                taskstatus.textContent = "تتم المعالجة الان"+result_data['donetasklist'].length+"/"+result_data['tasklist'].length;
            }else{
                taskstatus.textContent = "تتم المعالجة الان";
            }
            sleep(100);
            get_info();
            return;
        }
        taskstatus.textContent = "اكتمل :"+result_data['donetasklist'].length+"/"+result_data['tasklist'].length;
        show_info(result_data);
        // النتائج غير متسقة وتستمر في التحديث
        if(result_data['donetasklist'].length!=result_data['tasklist'].length){
            get_info();
        }
        return;
    });
}
get_info();